index.bs

```diff
@@ -2725,6 +2769,284 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Return [=success=] with data null.
 
+</div>
+#### The browsingContext.locateNodes Command ####  {#command-browsingContext-locateNodes}
+
+The <dfn export for=commands>browsingContext.locateNodes</dfn> command returns a
+list of all nodes matching the specified locators.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browsingContext.LocateNodes = (
+        method: "browsingContext.locateNodes",
+        params: browsingContext.LocateNodesParameters
+      )
+
+      browsingContext.LocateNodesParameters = {
+         context: browsingContext.BrowsingContext,
+         locators: [ + browsingContext.Locator ],
+         ? startNode: script.SharedReference,
+         ? waitTimeout: js-uint
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        browsingContext.LocateNodesResult = {
+            context: browsingContext.BrowsingContext,
+            nodes: [ * script.RemoteNodeValue ]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="locate nodes using CSS">
+To <dfn>locate nodes using CSS</dfn> with given |context|, |context nodes|,
+|selector| and <var ignore>session</var>:
+
+1. Let |returned nodes| be an empty array.
+
+1. For each |context node| of |context nodes|:
+
+  1. Let |elements| be the result of calling querySelectorAll() on |context| with |context node|
+     as this and |selector| as the argument. If this causes an exception to be thrown,
+     return [=error=] with [=error code=] [=invalid selector=]. 
+
+  1. Append |elements| to |returned nodes|
+
+1. Return |returned nodes|.
+
+</div>
+
+<div algorithm="locate nodes using XPath">
+To <dfn>locate nodes using XPath</dfn> with given |context|, |context nodes|
+|selector| and <var ignore>session</var>:
+
+1. Let |returned nodes| be an empty array.
+
+1. For each |context node| of |context nodes|:
+
+   1. Let |evaluateResult| be the result of calling evaluate on |context|, with arguments |selector|,
+      |context node|, null, ORDERED_NODE_SNAPSHOT_TYPE, and null.
+
+   1. Let |index| be 0.
+
+   1. Let |length| be the result of getting the property "snapshotLength" from
+      |evaluateResult|. If this throws an XPathException return [=error=] with
+      [=error code=] [=invalid selector=], otherwise if this throws any other
+      exception return return [=error=] with [=error code=] [=unknown error=].
+
+   1. Repeat, while |index| is less than |length|:
+
+      1. Let |node| be the result of calling snapshotItem with evaluateResult
+         as this and index as the argument.
+
+      1. If |node| is not an element return [=error=] with [=error code=]
+         [=invalid selector=].
+      
+      1. Append |node| to |returned nodes|.
+      
+      1. Increment |index| by 1. 
+
+1. Return |returned nodes|
+
+</div>
+
+<div algorithm="locate nodes using text">
```

Comment:
does this locator exist in WebDriver Classic? Is it the one called Link text?

Reply:
It does not, but is included to be able to construct "link text" and "partial link text." The intended payload for that would be something like:

```json
{
  "context": "some context",
  "locators": [
    {
      "type": "css",
      "value": "a"
    },
    {
      "type": "text"
      "value": "desired link text",
      "match": "full"
    }
  ]
}
```
Where `match` would be `full` for find by link text, and `partial` for find by partial link text.

Reply:
@jgraham The concrete use case for the `TextLocator` is "find some element with tag other than `<a>` by its visible text. Once we accept that's a legitimate use case (and I'm willing to believe that it is, so strongly that I'm willing to abandon this PR and let someone else author it), then replicating "find element by [partial] link text" makes sense to define as a combination of these primitives.

Reply:
@jgraham I've removed the locator chaining from this PR.